bukkit/src/main/java/com/viaversion/viaversion/bukkit/providers/BukkitPickItemProvider.java

```diff
@@ -187,20 +185,19 @@ static BlockToItem build() {
                     }
                     return item;
                 };
-            } else if (PaperViaInjector.hasMethod(Material.class, "isItem")) {
-                return (block, includeData) -> new ItemStack(block.getType(), 1);
+            } else if (LegacyBlockToItem.getInstance() != null) {
+                return (block, includeData) -> LegacyBlockToItem.getInstance().blockToItem(block);
             } else {
-                LegacyBlockToItem legacy = LegacyBlockToItem.getInstance();
-                return legacy != null ? (block, includeData) -> legacy.blockToItem(block) : (b, i) -> null;
+                return (block, includeData) -> new ItemStack(block.getType(), 1, (short) 0, block.getData());
```

Comment:
i guess this is a bug in my earlier pr, but this branch (which you've moved down) should be checking `block.getType().isItem()` before returning the itemstack with the blocktype, which is why it was checking for the existance of isItem to begin with

so i guess the new code should read something like:
```
else if (legacy != null):
  LegacyBlockToItem legacy = LegacyBlockToItem.getInstance();
  return (block, includeData) -> legacy.blockToItem(block);
else if (has method "isItem"):
  return (block, includeData) -> block.getType().isItem() ? new ItemStack(...) : null;
else
  return (b, i) -> null;
```
also note, you should extract legacy out of the lambda so you can save on the call each time the lambda is invoked

Reply:
Okay, please take a look at the newest version.